### PR TITLE
kubeadm: add directory struct to hold dir paths

### PR DESCRIPTION
**Pull request by padlar**
refactor setting up various k8s directory paths at a single location
fixes # kubernetes/kubeadm#1177

### CODE REVIEW

**Comment by padlar** [comment]
I signed it

File: cmd/kubeadm/app/cmd/init.go [hunk 1]
```diff
@@ -127,9 +127,38 @@ type initData struct {
...
 	dryRun                bool
 	ignorePreflightErrors sets.String
-	certificatesDir       string
 	dryRunDir             string
 	client                clientset.Interface
+	dirPaths              *dirPaths
+}
+
+// dirPaths stores all the directory paths
+type dirPaths struct {
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/init.go]: should be just called `paths` or `directories`.

File: cmd/kubeadm/app/cmd/init.go [hunk 2]
```diff
@@ -340,43 +365,6 @@ func (d initData) IgnorePreflightErrors() sets.String {
 	return d.ignorePreflightErrors
 }
 
-// CertificateWriteDir returns the path to the certificate folder or the temporary folder path in case of DryRun.
-func (d initData) CertificateWriteDir() string {
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/init.go]: if these getters are removed, how would other packages access the private struct members?

**Review by neolit123 — COMMENTED** [review]
thanks for the attempt at this @padlar 
added some comments.

**Comment by neolit123** [comment]
please sign the CLA and address review comments.
thanks!

**Comment by neolit123** [comment]
hi, we did a big refactor in init.go and this is not longer need.
feel free to send more PRs when you can!

thanks.

**Pull request closed without merging**